azurerm/internal/services/network/data_source_network_service_tags.go

```diff
@@ -0,0 +1,105 @@
+package network
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+)
+
+func dataSourceNetworkServiceTags() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNetworkServiceTagsRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"location": azure.SchemaLocation(),
+
+			"service": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"location_filter": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				StateFunc:        azure.NormalizeLocation,
+				DiffSuppressFunc: azure.SuppressLocationDiff,
+			},
+
+			"address_prefixes": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func dataSourceNetworkServiceTagsRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.ServiceTagsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	res, err := client.List(ctx, location)
+	if err != nil {
+		return err
+	}
+
+	if res.Values == nil {
+		return errors.New("unexpected nil value for service tag information")
+	}
+
+	service := d.Get("service").(string)
+	locationFilter := d.Get("location_filter").(string)
+
+	for _, sti := range *res.Values {
+		if sti.Name == nil || !isServiceTagOf(*sti.Name, service) {
+			continue
+		}
+
+		if props := sti.Properties; props != nil {
+			if *props.Region == locationFilter {
```

Comment:
How will this work if the location filter is empty? if i understand correctly some services are "global" so in those cases is Region empty? (could we get a comment to that affect if so) while others are regional and will have a location?

Reply:
we also need to nil check Region & normalize it when comparing with locationfilter

Reply:
I fixed normalization of the Region prop too. But from my understanding you get ALL address prefixes if you leave the location_filter out. Otherwise you get only the region you requested.

Reply:
👌 however we still need to nilcheck region beforedereferencingg